rails-snow-queen/app/assets/javascripts/map.js

```diff
@@ -0,0 +1,110 @@
+class Map {
+	constructor(config) {
+    this.map = this.initMap();
+    this.drawingManager = this.initDrawingManager();
+    this.removeControl = this.initRemoveControl();
+    this.geocoder = this.initGeocoder();
+    this.marker = null;
+    this.geocodedAddress = null;
+    this.polygons = [];
+    this.onGeocodingResponse = config.onGeocodingResponse;
+    this.onPolygonsCreated = config.onPolygonsCreated;
+
+    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
+    this.handleGeocodingResponse = this.handleGeocodingResponse.bind(this);
+    this.handlePolygonCreated = this.handlePolygonCreated.bind(this);
+    this.handleRemovePolygon = this.handleRemovePolygon.bind(this);
+    this.addListeners()
+  }
+  
+  initMap() {
+    return new google.maps.Map(document.getElementById('map'), {
+      zoom: 14,
+      center: {lat: 49.2860, lng: -122.8130},
+      mapTypeId: google.maps.MapTypeId.SATELLITE,
+      tilt:0
+    });
+  }
+  
+  initDrawingManager() {
+    const drawingOptions = {
+      fillColor: '#2DC1D6',
+      fillOpacity: 0.3,
+      strokeWeight: 5,
+      strokeColor: '#2DC1D6',
+      clickable: false,
+      // editable: true,
+      zIndex: 1
+    }
+    return new google.maps.drawing.DrawingManager({
+      drawingMode: google.maps.drawing.OverlayType.POLYGON,
+      drawingControl: true,
+      drawingControlOptions: {
+        position: google.maps.ControlPosition.TOP_CENTER,
+        drawingModes: ['polygon', 'rectangle']
+      },
+      map: this.map,
+      // markerOptions: {icon: 'https://developers.google.com/maps/documentation/javascript/examples/full/images/beachflag.png'},
+      // markerOptions: {icon: 'snow_flake.png'},
+      polygonOptions: drawingOptions,
+      rectangleOptions: drawingOptions
+    })
+  }
+  
+  initRemoveControl() {
+    const removeControlDiv = document.createElement('div');
+    removeControlDiv.classList.add("map-btn")
+    removeControlDiv.title = 'Click to remove selected area from the map';
+    removeControlDiv.innerHTML = 'Remove last';
```

Comment:
Now that you've [added jQuery as a dependency](https://github.com/rosadiaz/snow_queen_v2/pull/1/files/4a57a1d4c8f2c27b1e9d949b2e0df9176b93d071..b7a1888932625c0826cf7c674f654a255430bee0#diff-950f5660d1465b178f38119caca7295cR13), I would do DOM manipulation with it. I find it much "nicer" than plain JS. However, that _does_ tie you to jQuery which may or may not be necessary later.
```javascript
const $removeControlDiv = $('<div/>', {
  text: 'Remove last',
  'class': 'map-btn', 
  title: 'Click to remove selected area from the map'
});

const node = $removeControlDiv.get(0); // $removeControlDiv is a jQuery wrapped object, must get the bare node to pass to google API
node.index = 1; 
this.map.controls[google.maps.ControlPosition.TOP_CENTER].push(node);

return $removeControlDiv;
```


Similarly

Reply:
If you're planning to switch to React, the need for jQuery dom manipulation could be completely eliminated! I guess that's the next step 🥇

Reply:
Since I'm planning on Reactifying it in the future I'll leave with DOM manipulation for now